rfcs/convergence/remove-commons.md

```diff
@@ -0,0 +1,135 @@
+# RFC: Remove `___Commons` types
+
+---
+
+Contributors: @ecraig12345
+
+## Summary
+
+To make our components' public API contracts clearer to users:
```

Comment:
As a new contributor I'd say this change will make things clearer for me. It's entirely possible that I'm missing something but the `___Commons` types don't seem to provide any utility -- they are just lists of what is common between props, state and context